drivers/dma/dma_stm32.h

```diff
@@ -45,6 +45,15 @@ struct dma_stm32_config {
 	u32_t base;
 };
 
+#ifdef CONFIG_DMA_STM32_V1
+/* from DTS the dma stream id is in range 0..<dma-requests>-1 */
+#define STREAM_OFFSET 0
+#else
+/* from DTS the dma stream id is in range 1..<dma-requests> */
+/* so decrease to set range from 0 from now on */
+#define STREAM_OFFSET 1
+#endif /* CONFIG_DMA_STM32_V1 */
+
```

Comment:
It seems to me that for L4 is 1 to 7 (not 8). Correct?
For F4 is 0 to 7.

Reply:
Yes, this is very confusing in the stm32 ref manuals.
For F4 (DMA V1) requests are called 'channels' 
For L4 (DMA V2) channels (what DMA V1 calls 'streams') have several requests.
I tried to format in the bindings the channel (or stream for DMA_V1) in the first position and the request (or channel in DMA_V1) in the second position, based on the table given by the ref Man ( whatever the numbering from 0 to 7 or from 1 to 8 is).
For example on L476:
 the SPI1_TX  on Channel 3 (from 1 to 8) has request 1 (from 0 to 7)
 the SPI1_RX on Channel 2 (from 1 to 8) has request 1 (from 0 to 7) 
For example on F411:
 the SPI1_TX  on stream 5 (from 0 to 7) has request channel 3 (from 0 to 7)  
 the SPI1_RX on stream 2 (from 0 to 7) has request channel  3 (from 0 to 7)

Reply:
YEs, very confusing.

Reply:
OK, I think it is correct in this way.